resource/graph_node.go

```diff
@@ -11,6 +11,24 @@ import (
 	"go.viam.com/rdk/logging"
 )
 
+// NodeState captures the configuration lifecycle state of a resource node.
+type NodeState uint8
+
+const (
+	// NodeStateUnconfigured denotes a newly created resource.
+	NodeStateUnconfigured NodeState = iota
+
+	// NodeStateConfiguring denotes a resource is being configured.
+	NodeStateConfiguring
+
+	// NodeStateReady denotes a resource that has been initialized and is not being
+	// configured or removed. A resource in this state may be unhealthy.
+	NodeStateReady
+
+	// NodeStateRemoving denotes a resource is being removed from the resource graph.
+	NodeStateRemoving
+)
+
 // A GraphNode contains the current state of a resource.
 // It starts out as either uninitialized, unconfigured, or configured.
```

Comment:
should we also add `uninitialized` or is that an impossible state?

Reply:
it's not clear to me how this state would be different from `unconfigured` (i.e. the default state) as currently proposed - right now i'm using `unconfigured` to capture everything before `configuring`

Reply:
can you update the docstring then?

Reply:
somewhat related, it might be worth having an "unknown" state as the default/sentinel value. this value would always be treated as an error